drop-transfer/src/check.rs

```diff
@@ -7,51 +7,91 @@ use tokio_util::sync::CancellationToken;
 use crate::{protocol, service::State, tasks::AliveGuard, IncomingTransfer, Transfer};
 
 pub(crate) fn spawn(
+    refresh_trigger: tokio::sync::watch::Receiver<()>,
     state: Arc<State>,
     xfer: Arc<IncomingTransfer>,
     logger: Logger,
     guard: AliveGuard,
     stop: CancellationToken,
-) -> tokio::task::JoinHandle<()> {
+) {
     let id = xfer.id();
-    let job = run(state, xfer, logger.clone());
 
     tokio::spawn(async move {
         let _guard = guard;
+        let mut refresh_trigger = refresh_trigger;
+
+        let logger_clone = logger.clone();
+        let task = async move {
+            let logger = logger_clone;
+            loop {
+                let cf = run(&state, xfer.clone(), &logger).await;
+
+                if cf.is_break() {
+                    break;
+                }
+
+                #[allow(clippy::let_underscore_must_use)]
+                let _ = refresh_trigger.changed().await;
+            }
+        };
 
         tokio::select! {
             biased;
 
             _ = stop.cancelled() => {
-                debug!(logger, "Check job stop for transfer: {id}");
+                debug!(&logger, "stop checking job for: {}", id);
             },
-            _ = job => (),
+            _ = task => ()
         }
-    })
+    });
 }
 
-async fn run(state: Arc<State>, xfer: Arc<IncomingTransfer>, logger: Logger) {
-    loop {
-        tokio::time::sleep(drop_config::ALIVE_CHECK_INTERVAL).await;
+async fn run(state: &State, xfer: Arc<IncomingTransfer>, logger: &Logger) -> ControlFlow<()> {
```

Comment:
```suggestion
async fn run(state: &State, xfer: &IncomingTransfer, logger: &Logger) -> ControlFlow<()> {
```

Reply:
This one is not actually that easy, downstream calls require it to be Arc

Reply:
So then
```suggestion
async fn run(state: &State, xfer: &Arc<IncomingTransfer>, logger: &Logger) -> ControlFlow<()> {
```
That should reduce the amount of cloning

Reply:
I don't personally get used to passing Arcs as references, but this worked